File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/ItemPromotionSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface ItemPromotionSearchModel extends BaseSearchModel {
    vendorId: number | null;
    vendorCode:string;
    itemId: number | null;
    itemCode:string;
    discount: number | null;
    validTo: Date | null;
    validFrom: Date | null;
}

File: SofiForce.Models/Models/SearchModels/ItemPromotionSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class ItemPromotionSearchModel : BaseSearchModel
    {

		public Int32? VendorId { get; set; }
		public Int32? ItemId { get; set; }

		public decimal? Discount { get; set; }
		public DateTime? ValidTo { get; set; }
		public DateTime? ValidFrom { get; set; }



	}
}